Added new kb article integrating-navigationview-with-maui-navigation (#1075)
telerik/maui-docs: knowledge-base/integrating-navigationview-with-maui-navigation.md

---
title: Integrating NavigationView with MAUI Navigation
description: Learn how to integrate Telerik UI for .NET MAUI NavigationView with MAUI's navigation system.
type: how-to
page_title: How to Use NavigationView with MAUI's Navigation System
slug: integrating-navigationview-with-maui-navigation
tags: navigationview, .net maui, navigation, ui for .net maui
res_type: kb
---

## Description

Understanding how Telerik UI for .NET MAUI [NavigationView](https://docs.telerik.com/devtools/maui/controls/navigationview/overview) integrates with the .NET MAUI Navigation system is crucial for developing seamless navigation within your application. This knowledge base article also answers the following questions:

- How do I navigate between pages using NavigationView in a .NET MAUI application?
- Can NavigationView work with MAUI NavigationPage for navigating through pages?
- How to handle selection changes in NavigationView for page navigation?

## Environment

| Versions | Product | Author | 
| --- | --- | ---- | 
| 9.0.0 | Telerik UI for .NET MAUI NavigationView | [Dobrinka Yordanova](https://www.telerik.com/blogs/author/dobrinka-yordanova) | 

## Solution

> To integrate Telerik UI for .NET MAUI NavigationView with MAUI navigation, follow these steps:

Reviewer: ```suggestion
To integrate Telerik UI for .NET MAUI NavigationView with MAUI navigation, first consider the difference between the two: Telerik MAUI `RadNavigationView` is a control for displaying navigation items (views, control, etc.) and does not inherently manage page navigation. In contrast, MAUI `NavigationPage` manages a stack of pages, facilitating navigation and user experience.
```

> 1. **Understand the Difference**: Telerik MAUI `RadNavigationView` is a control for displaying navigation items (views, control, etc.) and does not inherently manage page navigation. In contrast, MAUI `NavigationPage` manages a stack of pages, facilitating navigation and user experience.

Reviewer: I don't think that we could use this as an actual step in a procedure. I suggest making it part of the explanation.

```suggestion
```

2. **Implement Navigation Logic**: Use the `SelectionChanged` event of the `NavigationView` to implement custom navigation logic. When a navigation item is selected, navigate to the corresponding page using the `Navigation.PushAsync` method.

> ### NavigationView Implementation

Reviewer: ```suggestion
### NavigationView Implementation

Use the `SelectionChanged` event of the `NavigationView` to implement custom navigation logic. When a navigation item is selected, navigate to the corresponding page using the `Navigation.PushAsync` method.
```
